core/src/subgraph/instance_manager.rs

```diff
@@ -358,13 +364,10 @@ impl SubgraphInstanceManager {
         let block_filter = EthereumBlockFilter::from_data_sources(&manifest.data_sources);
         let start_blocks = manifest.start_blocks();
 
-        // Identify whether there are templates with call handlers or
+        // Identify whether there are mappings with call handlers or
         // block handlers with call filters; in this case, we need to
-        // include calls in all blocks so we cen reprocess the block
-        // when new dynamic data sources are being created
-        let templates_use_calls = templates.iter().any(|template| {
-            template.has_call_handler() || template.has_block_handler_with_call_filter()
-        });
+        // include calls in all blocks
+        let include_calls_in_blocks = manifest.requires_traces();
```

Comment:
Here too I would do something like
```rust
let include_calls_in_blocks = manifest
    .required_ethereum_capabilities()
    .contains(&NetworkCapability::Archive);
```

Reply:
On the other hand, using `requires_traces()` / `required_ethereum_capabilities()` for determining whether to include calls doesn't seem right. I think the check before was easier to understand and less likely to lead to an incorrect result. Just imagine we add something else that also requires traces. Suddenly we'd include calls by accident even though the feature may not require them at all.

Reply:
This is still valid.

Reply:
Actually. I'm ok with this.